src/main/java/com/minecolonies/coremod/entity/ai/util/RecipeStorage.java

```diff
@@ -56,7 +57,7 @@ public RecipeStorage(final List<ItemStack> input, final int gridSize, final Item
      */
     public List<ItemStack> getInput()
     {
-        return InventoryUtils.hardCopyListOfItemStacks(input);
+        return new ArrayList<>(input);
```

Comment:
No need to make a new list, no one should be adding or removing anything from this list.
Actually you should change the constructor to be:
```
         this.input = Collections.unmodifiableList(input);
         this.primaryOutput = primaryOutput;
         this.secondaryOutput = Collections.unmodifiableList(Arrays.asList(secondaryOutput));
         this.gridSize = gridSize;
```